### ISSUE TITLE
minified name in react devtools

### ISSUE THREAD

**Issue by Janpot** [leefsmp/Re-Flex#13]
<img width="572" alt="screen shot 2017-05-20 at 16 37 18" src="https://cloud.githubusercontent.com/assets/2109932/26276618/f23a3c0a-3d7a-11e7-9dd0-6373fa99fad3.png">

I think you need to set `displayName` on your components for this to work. 
http://babeljs.io/docs/plugins/transform-react-display-name/
https://github.com/facebook/react-devtools

**Comment by Janpot** [comment]
Ok, strange. It's supposed to be part of the [`react` preset](https://github.com/leefsmp/Re-Flex/blob/master/webpack/lib/production.webpack.config.js#L65).
I'm wondering if it gets minified away somehow.

**Comment by leefsmp** [comment]
I tried [babel-plugin-add-react-displayname](https://github.com/opbeat/babel-plugin-add-react-displayname) but it doesn't help. I think it does get minified but dunno how to prevent that ...

**Comment by Janpot** [comment]
@leefsmp  Ok, it seems `displayName` is only for when you do `React.createClass`. The issue here is most likely due to the minification. I'm suspecting it's also the cause of the still remaining proptypes warning, although I haven't been able to figure out what's happening there exactly yet.

Are you open to some optimization of the build process? I've been looking around a bit.
Ideally a library would expose a commonjs version through the package `main` field and an ES6 version through the package `module` field. Most components I come across also offer an umd module as well, which you are already building, only you are exposing it as the package `main`.
Most libraries seem to build their commonjs and ES6 versions with babel and their umd with webpack.

What makes converting your library non-trivial is that you use the style loader for your CSS. The libraries I usually come across expose their styles through a separate CSS file, thus leaving the way to include CSS up to the consumer. Since most people already use webpack to build their react apps this is easy to include. Are you willing to change the way styles are loaded by this library? In any case we could also keep the umd bundle as it is but build a separate css for the commonjs and ES6 versions of this library.

here's an example of such app https://github.com/bvaughn/react-virtualized/blob/master/package.json

I can look into this if you are open to this approach.

btw, replacing the `main` field with commonjs version is potentially a breaking change

**Comment by Janpot** [comment]
I made a prototype of this in https://github.com/Janpot/Re-Flex/commit/9b9960df7363774348f307e39d1c12b27dfea0bb
can be tested from [`@janpotoms/react-reflex`](http://npmjs.com/package/@janpotoms/react-reflex)

```js
import { ReflexContainer, ReflexSplitter, ReflexElement } from '@janpotoms/react-reflex';
import '@janpotoms/react-reflex/styles.css'
```

advantages:
1. build size dropped ~10K for my test project. This project also imports `react-measure` and `prop-types` for other purposes. `babel-runtime` is also deduplicated. This is because dependencies can be shared and babel can treeshake code that isn't used
2. the proptypes warning is gone. I think that was caused by the heavy minification.
3. Errors and devtools have proper values for the components (this issue, also caused by heavy minification)
4. can still serve a umd bundle for people that need it
5. I also added a prepublish step and removed the build files from git

disadvantages
1. breaking change
2. separate import for CSS (I personally prefer this but I can understand if it's a disadvantage for you). Maybe there's ways to make it work as it works now.
